cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/submit/MasterEnvironmentSparkSubmitter.java

```diff
@@ -141,6 +152,22 @@ protected void onCompleted(boolean succeeded) {
     } else {
       sparkExecutionService.shutdownNow();
     }
+    try {
+      if (sparkDriverWatcherThread instanceof AutoCloseable) {
+        ((AutoCloseable) sparkDriverWatcherThread).close();
+      }
+    } catch (Exception e) {
+      LOG.error("Error while closing spark driver watcher thread.", e);
+    }
+  }
+
+  @Override
+  protected boolean hasSparkDriverSucceeded() throws Exception {
```

Comment:
this name makes it sound like it returns immediately, but it can actually block for a long time. It would be better to rename it to something more descriptive

Reply:
still have this comment

Reply:
Ah yea, it didnt get synced.. Fixed